code/modules/interaction_tests/test_puppeteer.dm

```diff
@@ -0,0 +1,49 @@
+/datum/test_puppeteer
+	var/mob/living/carbon/puppet
+	var/list/tracked_atoms
+
+/datum/test_puppeteer/New(carbon_type, turf/initial_location)
+	if(!initial_location)
+		initial_location = locate(179, 136, 1) // Center of admin testing area
+	puppet = new carbon_type(initial_location)
+	var/datum/mind/new_mind = new("interaction_test_[rand(1, 999999)]")
+	new_mind.transfer_to(puppet)
+
+/datum/test_puppeteer/proc/spawn_obj_in_hand(obj_type)
```

Comment:
Might want a variant of this function that has varargs for custom initialization?

Reply:
I'll be honest, I have no idea how to properly pass varargs in here and have them properly appended to the allocate call. Could you take a look at the new version and give me a hint?